CLOUDFRONT.md

```diff
@@ -58,51 +56,60 @@ php admin/cli/scheduled_task.php --execute='\tool_objectfs\task\generate_status_
-6. _Viewer protocol policy_: Choose _Redirect HTTP to HTTPS_.
-7. _Allowed HTTP methods_: Choose _GET, HEAD, OPTIONS_ and tick _OPTIONS_ under _Cache HTTP methods_.
-8. _Restrict viewer access_: Choose _Yes -> Trusted signer -> Self_.
-9. _Cache key and origin requests_: Choose _Legacy cache settings_.
-10. _Legacy cache settings -> Headers_: Choose _Include the following headers_ and add _Origin_, _Access-Control-Request-Method_, _Access-Control-Request-Headers_ headers from the dropdown menu.
-11. _Legacy cache settings -> Query strings_: Choose _All_.
-12. Click _Create distribution_.
-13. Navigate to [https://console.aws.amazon.com/cloudfront/v3/home?region=ap-southeast-2#/distributions].
-14. Confirm, that _Status_ is _Enabled_ and _Last modified_ is changed from _Deploying_ to the date the distribution was created.
-15. Open your distribution.
-16. Write down _Distribution domain name_ (with https://).
-> Note: If you have already setup Moodle behind a CloudFront distribution, it is also possible to use that same CloudFront distribution to serve files from objectfs. In this scenario, a specific prefix in the URL path directs traffic to the S3 Bucket (moodle.domain/objectfs/ for example). To achieve that, use the key_prefix option to add a prefix on your Bucket, and configure a second Origin on your existing CloudFront distribution that points to your Bucket. Setup a Behavior that uses that new Origin with the same prefix as the one you used as key_prefix in your Bucket. Follow all other instructions.
-
-### Generate keys
-1. Make a directory _$CFG->dataroot . '/objectfs/'_.
-2. Make it readable and writable:
+### Generate CloudFront keys
+1. Generate an RSA key pair with a length of 2048 bits:
 ```
-chmod 777 objectfs
-```
-3. Generate an RSA key pair with a length of 2048 bits:
-```
-cd objectfs/
 openssl genrsa -out cloudfront.pem 2048
-chmod 777 cloudfront.pem
 ```
-4. Extract the public key:
+2. Extract the public key:
 ```
 openssl rsa -pubout -in cloudfront.pem -out public_key.pem
 ```
-5. Navigate to [https://console.aws.amazon.com/cloudfront/v3/home#/distributions].
-6. In the navigation menu, choose _Public keys_.
-7. Click _Create public key_.
-8. Enter key name.
-9. Enter key value. Use the following command to get the public key:
+3. Navigate to https://console.aws.amazon.com/cloudfront/v3/home#/publickey.
+4. Click _Create public key_.
+5. Enter key name.
+6. Enter key value. Use the following command to get the public key:
 ```
 cat public_key.pem
 ```
-10. Click _Create public key_.
-11. Write down key ID from the [https://console.aws.amazon.com/cloudfront/v3/home#/publickey] page.
+7. Click _Create public key_.
+8. Write down key ID from the https://console.aws.amazon.com/cloudfront/v3/home#/publickey page.
+9. Navigate to https://console.aws.amazon.com/cloudfront/v3/home#/keygrouplist
+10. Create a key group and select the public key created previously
```

Comment:
Yeah, this bit me and I still can feel the pain... Weird it didn't work with `self`

Reply:
It is still possible to use `self`, but that is a legacy method (and requires root account access) and using key groups is now recommended instead.